tests/utils/test_cache_utils.py

```diff
@@ -713,3 +716,210 @@ def test_hybrid_cache_exportability(self):
             dynamic_shapes=dynamic_shapes,
             strict=False,
         )
+
+
+class SyntheticCacheTest(unittest.TestCase):
+    """
+    Synthetic tests for StaticCache, SlidingWindowCache, and HybridCache.
+    Uses window_size=4, max_cache_len=4 for all tests.
+    """
+
+    def setUp(self):
+        """Set up common configuration for all tests."""
+        self.window_size = 4
+        self.max_cache_len = 4
+        self.config = SimpleNamespace(
+            num_hidden_layers=1,
+            num_key_value_heads=1,
+            num_attention_heads=1,
+            head_dim=1,
+            hidden_size=1,
+            sliding_window=self.window_size,
+            sliding_window_pattern=2,  # Example pattern
+        )
+
+    def _extract_both_caches(self, cache):
+        """Helper to extract flattened key and value states from the cache."""
+        # Assumes layer 0, batch 0, head 0
+        k = cache.key_cache[0][0, 0, :, 0].tolist()
+        v = cache.value_cache[0][0, 0, :, 0].tolist()
+        return k, v
+
+    def test_static_cache_within_bounds(self):
+        """
+        Test StaticCache preserves values at written positions within bounds.
+        Example (max_cache_len=4):
+        step 1 (pos 0): [1.0, 0.0, 0.0, 0.0]
+        step 2 (pos 1): [1.0, 2.0, 0.0, 0.0]
+        step 3 (pos 2): [1.0, 2.0, 3.0, 0.0]
+        step 4 (pos 3): [1.0, 2.0, 3.0, 4.0]
+        """
+        static_cache = StaticCache(config=self.config, max_batch_size=1, max_cache_len=self.max_cache_len)
+        expected_state = [0.0] * self.max_cache_len
+
+        for step in range(1, self.max_cache_len + 1):  # Test up to cache capacity
+            pos_idx = step - 1
+            value = float(step)
+
+            static_cache.update(
+                key_states=torch.tensor([[[[value]]]]),
+                value_states=torch.tensor([[[[value]]]]),  # Use same value for simplicity
+                layer_idx=0,
+                cache_kwargs={"cache_position": torch.tensor([pos_idx])},
+            )
+
+            expected_state[pos_idx] = value
+            stored_keys, stored_values = self._extract_both_caches(static_cache)
+
+            self.assertEqual(stored_keys, expected_state, f"Static Key cache failed at step {step}")
```

Comment:
In these tests, I think it's preferable to hardcode the outputs (as opposed to generating them on the fly)

Then, instead of documenting the expected output in the docstring, we can simply look at the variable with hardcoded outputs